sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVHeaderChecker.scala

```diff
@@ -75,22 +75,24 @@ class CSVHeaderChecker(
             // scalastyle:on caselocale
           }
           if (nameInHeader != nameInSchema) {
+            // scalastyle:off line.size.limit
             errorMessage = Some(
```

Comment:
It seems that only the message content of the assertion has changed. Would it be more appropriate to directly fix the corresponding test case?

Reply:
- The original message `format` is as follows:
```
Map("msg" -> "CSV header does not conform to the schema.
 Header: columnA, columnB
 Schema: columnB, columnA
Expected: columnB but found: columnA
CSV source: [value: string]")
```

- After https://github.com/apache/spark/pull/45904, the message `format` is as follow: 
```
Map("msg" -> "CSV header does not conform to the schema.
 Header: columnA, columnB
 Schema: columnB, columnA
Expected: columnB
but found: columnA
CSV source: [value: string]")
```
If the corresponding UT is modified, it can be passed, but the modified format after https://github.com/apache/spark/pull/45904 seems to go against the `original intention` and is not particularly `reasonable`

Reply:
And also it be as a message value for exception error classes:
https://github.com/apache/spark/blob/070461cc673c3fc910e66d1cbf628632b558b48c/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVHeaderChecker.scala#L102

Reply:
I accept this change because I can't think of a better way to fix it. Let's wait for suggestions from others.